arches_for_science/trigger_functions.py

```diff
@@ -0,0 +1,184 @@
+            ELSE 'description'
+        END
+        INTO descriptor_key;
+
+        -- Resolve node values to localized strings
+        FOR alias_with_separators IN SELECT UNNEST(REGEXP_MATCHES(this_template, '<\w+>', 'g'))
+        LOOP
+            SELECT TRIM(BOTH '<>' FROM alias_with_separators) INTO node_alias;
+
+            SELECT nodeid, nodegroupid INTO node_for_alias, nodegroup_for_alias
+                FROM nodes
+                WHERE alias = node_alias
+                AND graphid = graph;
+
+            SELECT tiledata ->> node_for_alias::text INTO localized_string_node_value
+                FROM tiles
+                WHERE resourceinstanceid = resourceid
+                AND nodegroupid = nodegroup_for_alias
+                AND sortorder = 0
+                LIMIT 1;
+
+            -- Replace template with localized string
+            FOR lang, inner18n_obj_for_lang IN SELECT * FROM jsonb_each(localized_string_node_value) 
+            LOOP
+                -- Initialize language key if missing
+                IF localized_result_all -> lang IS NULL THEN
+                    SELECT jsonb_set(
+                        localized_result_all,
+                        ARRAY[lang],
+                        JSONB_OBJECT('{}')
+                    ) INTO localized_result_all;
+                END IF;
+
+                -- Retrieve the current working value, or start from the template
+                SELECT
+                CASE
+                    WHEN localized_result_all -> lang -> descriptor_key IS NULL
+                        THEN this_template
+                    ELSE TRIM((localized_result_all -> lang -> descriptor_key)::text, '\"')
+                END
+                INTO working_string;
+
+                SELECT REPLACE(
+                    working_string,
+                    alias_with_separators,
+                    TRIM((inner18n_obj_for_lang -> 'value')::text, '"')
+                )
+                INTO resolved_node_value_for_lang;
+
+                -- Update the working value
+                SELECT jsonb_set(
+                    localized_result_all,
+                    ARRAY[lang, descriptor_key],
+                    TO_JSONB(resolved_node_value_for_lang)
+                ) INTO localized_result_all;
+
+                IF descriptor_key = 'name' THEN
+                    SELECT jsonb_set(
+                        localized_result_name_only,
+                        ARRAY[lang],
+                        TO_JSONB(resolved_node_value_for_lang)
+                    ) INTO localized_result_name_only;
+                END IF;
+
+            END LOOP;
+
+        END LOOP;
+
+    END LOOP;
+
+    UPDATE resource_instances
+    SET descriptors = localized_result_all, name = localized_result_name_only
+    WHERE resourceinstanceid = resourceid;
+
+END IF;
+END;
+RETURN NULL;
+"""
+
+#### Make two adjustments to create a very similar function that works on ALL resource instances.
```

Comment:
I wasn't thrilled about this string replacement to create the second trigger, but only two small places need to be edited, and this avoids duplicating a lot of code.

I considered f-strings, but I got nervous about dealing with escaping curly brackets. I'm open to it, though.